src/server/routes/auth/keypair/import.ts

```diff
@@ -0,0 +1,83 @@
+import { Keypairs, Prisma } from "@prisma/client";
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { insertKeypair } from "../../../../db/keypair/insert";
+import { createCustomError } from "../../../middleware/error";
+import {
+  KeypairAlgorithmSchema,
+  KeypairSchema,
+} from "../../../schemas/keypairs";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+
+const BodySchema = Type.Object({
+  publicKey: Type.String({
+    description:
+      "The public key of your keypair beginning with '-----BEGIN PUBLIC KEY-----'.",
+  }),
+  algorithm: KeypairAlgorithmSchema,
+});
+
+const ReplySchema = Type.Object({
+  result: Type.Object({
+    keypair: KeypairSchema,
+  }),
+});
+
+export async function importKeypair(fastify: FastifyInstance) {
+  fastify.route<{
+    Body: Static<typeof BodySchema>;
+    Reply: Static<typeof ReplySchema>;
+  }>({
+    method: "POST",
+    url: "/auth/keypair/import",
+    schema: {
+      summary: "Import public key",
+      description: "Import the public key for a keypair",
+      tags: ["Keypair"],
+      operationId: "import",
+      body: BodySchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: ReplySchema,
+      },
+    },
+    handler: async (req, res) => {
+      const { publicKey, algorithm } = req.body;
+
+      if (
+        !publicKey.startsWith("-----BEGIN PUBLIC KEY-----\n") ||
```

Comment:
is this always consistent? 
is there a library or better way to do the validation based off the algorithm rather than hardcoding the string?

Reply:
Eh yeah I don't like this either. I tested each of the RS*/ES*/PS* and they all started with `-----BEGIN PUBLIC KEY-----`.

Reply:
Actually this works quite well:
```
  function isWellFormedPublicKey(publicKeyString) {
    try {
      crypto.createPublicKey(publicKeyString);
      return true;
    } catch (e) {
      return false;
    }
  }
```